projects/sbb-esta/angular-business/src/lib/header/navbutton/navbutton.component.ts

```diff
@@ -29,52 +33,67 @@ export class NavbuttonComponent implements AfterViewInit, OnDestroy {
   @ViewChild('button', { static: true }) childButton: ElementRef;
 
   /**
-   * Observer to detect when an eventual dropdown attached to childButton is expanded or collapsed.
-   * Used to propagate change detection.
+   * Dropdown trigger on host, if present, for dropdown interaction and detection.
    */
-  private _attributeObserver: MutationObserver;
+  @ContentChild(DropdownTriggerDirective, { static: true })
+  dropdownTrigger: DropdownTriggerDirective;
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+  /**
+   * Returns whether childButton's dropdown is expanded.
+   */
+  private _isDropdownExpanded = false;
+  @HostBinding('class.sbb-navbutton-dropdown-expanded')
+  get isDropdownExpanded() {
+    return this._isDropdownExpanded;
+  }
+
+  /**
+   * Css width to apply when the dropdown is opened.
+   */
+  dropdownWidth = '200px';
+
+  /** @docs-private */
+  private _subscriptions: Subscription[] = [];
+
+  constructor(@SkipSelf() private _changeDetectorRef: ChangeDetectorRef, private _el: ElementRef) {}
```

Comment:
You should be able to optionally inject DropdownTriggerDirective: @Optional() private _dropdownTrigger: DropdownTriggerDirective instead of the ContentChild